src/rust/ensogl/src/display/scene.rs

```diff
@@ -120,23 +120,63 @@ pub enum Target {
 }
 
 impl Target {
+
+    /// Decode the symbol_id and instance_id that was encoded in the `fragment_runner`.
+    ///
+    /// See the `encode` method for more information on the encoding.
+    fn decode(pack1:u32, pack2:u32, pack3:u32) -> (u32,u32) {
+        let value1 = (pack1 << 4) + (pack2 >> 4);
+        let value2 = pack3 + ((pack2 & 0x000F) << 8);
+        (value1, value2)
+    }
+
+    /// Encode two u32 values into three u8 values.
+    /// This is the same encoding that is used in the `fragment_runner`.
+    /// This encoding is lossy.
```

Comment:
Please use all 100 chars wide in doc sentences.

Reply:
Plus, as I normally love your docs because they contain every information that is needed, here I'm missing some important info, namely info about max ID that could be encoded as `symbol_id` and `instance_id`. It should be told very explicitly. Moreover, when trying to encode too big id, an warning should be emitted to console by using our logger.